### PR TITLE
2024-08-28: @next changes

### PR DESCRIPTION
**Pull request by daviesgeek**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -38,20 +38,19 @@ services:
...
 
   caddy:
     image: caddy:latest
     ports:
       - 80:80
-      - 443:443
+      - 8443:443
     volumes:
       - ./caddy/data:/data/caddy
-      - ./Caddyfile:/etc/caddy/Caddyfile
```
Comment (kevinelliott) [inline docker-compose.yml:54]: I'm guessing just an oopsie?
docker-compose.yml --> Caddyfile.dev

File: frontend/src/slices/apiSlice.ts [hunk 2]
```diff
@@ -25,7 +23,9 @@ export const apiSlice = createApi({
     "Messages",
     "MqttMessages",
   ],
-  baseQuery: fetchBaseQuery({ baseUrl: `${apiBaseURI}/api/v1` }),
+  baseQuery: fetchBaseQuery({
+    baseUrl: `${window.location.origin}/api/v1`,
```
Comment (kevinelliott) [inline frontend/src/slices/apiSlice.ts:27]: This doesn't work if you are running the SPA and the API separately but not gated through nginx/caddy. Shouldn't this receive the correct base uri?

File: docker-compose.yml (hunk 1 above)
Comment (daviesgeek) [reply docker-compose.yml:54]: yeah, mb will revert

File: frontend/src/slices/apiSlice.ts (hunk 2 above)
Comment (daviesgeek) [reply frontend/src/slices/apiSlice.ts:27]: The Vite dev server will proxy this correctly since it's not actually nested under `/next`.

Comment (kevinelliott) [reply frontend/src/slices/apiSlice.ts:27]: Right, but run the production frontend on port 8000 and the api on port 9000, and do not front it with nginx or caddy. And you'll see what I'm talking about.

Comment (daviesgeek) [reply frontend/src/slices/apiSlice.ts:27]: yeah agree there, can we push this off for now until we don't have to deal with subpath issues? Else I think we can just use env vars to pass to the FE build: https://vitejs.dev/guide/env-and-mode

Comment (daviesgeek) [reply frontend/src/slices/apiSlice.ts]: @kevinelliott done! Also added CORS support to the uvicorn server

**Pull request merged**